blueocean-dashboard/src/main/js/PipelineRoutes.jsx

```diff
@@ -127,6 +125,27 @@ function onLeaveCheckBackground() {
 
 const trends = AppConfig.isFeatureEnabled('trends');
 
+function getActions(extensionPoint, params, callback) {
+    Extensions.store.getExtensions([ extensionPoint ], Extensions.Utils.sortByOrdinal, (actions = []) => {
+        const action = actions.map(a => new a()).find(action => action.name === params.action);
+        if (action) {
+            callback(null, action.component);
+        } else {
+            callback(null, NotFound);
+        }
+    });
+}
+
+function getRunDetailsAction({ params }, callback) {
```

Comment:
Find all extensions of the `jenkins.run.actions` extension point. Special consideration is taken for the default run details (changes, tests, artifacts) as they have associated telemetry callbacks.